### ISSUE TITLE
System.Runtime.InteropServices.RuntimeInformation package lacks type forwarding facade

### ISSUE THREAD

**Issue by AArnott** [dotnet/runtime#67063]
### Description

I have a .NET Framework 4.7.2 project that references mscorlib and System.Runtime.InteropServices.RuntimeInformation as follows:

- C:\NugetCache\VS.ExternalAPIs.NetFX.V4_7_2.1.0.18091101\ref\mscorlib.dll
- C:\Users\andarno\.nuget\packages\system.runtime.interopservices.runtimeinformation\4.0.0\ref\netstandard1.1\System.Runtime.InteropServices.RuntimeInformation.dll

Both of these assemblies define the 
System.Runtime.InteropServices.RuntimeInformation.Architecture type, leading to this compiler error whenever the source code references that type:

BTW: in preparing a repro for this, I noticed that SDK style projects do not suffer from this. Evidently they automatically get a reference to `C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.InteropServices.RuntimeInformation.dll`

Note the "Facades" part of the path. That reference evidently eclipses the dll by the same name that comes from the nuget package so it works fine. It's older projects that don't automatically reference this facade, but _do_ reference mscorlib and the nuget package that have the problem.

Most dotnet/runtime packages include facade assemblies where necessary. I feel this one is lacking it but should have it too.

### Reproduction Steps

This fails the build in our `src\vsproject\PackageAndDeploy\DeploymentService.UnitTests` project in the devdiv/VS repo.
I cannot repro with a stock template .NET Framework (old style) or SDK style project. Evidently they are smart enough to deliver the facade assembly from the .NET Framework 4.7.2 SDK. But this project is peculiarly dumb enough to not get that facade, but smart enough to honor transitive package references, which is how I found that no facade assembly in that package exists.

### Expected behavior

The project compiles fine because the System.Runtime.InteropServices.RuntimeInformation package should deliver a facade assembly for net471+ targeted projects.

### Actual behavior

> error CS0433: The type 'Architecture' exists in both 'System.Runtime.InteropServices.RuntimeInformation, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' and 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'

Now, what I expected to find when investigating this is that I was referencing the wrong System.Runtime.InteropServices.RuntimeInformation package version or ref assembly. I expected to find that there was a build of that assembly that used type forwarders to mscorlib instead of declaring the types itself. But in the latest 4.3.0 package version, the lib/net45 and ref/netstandard1.1 directories both contained assemblies that declares the type. 
[apisof.net shows](https://apisof.net/catalog/b04ab9f0-bb83-a610-2c88-243c8c427bbc) that mscorlib likely picked up its own copy of this type in net471. So I would expect the System.Runtime.InteropServices.RuntimeInformation package to include a lib/net471 directory that uses a .dll with type forwarders. But no such assembly exists.

As a result, the compiler produces an error when given two dll's that both declare the types.

### Regression?

It (presumably) works when the project targets older frameworks than net471.

### Known Workarounds

Adding this to the project resolves the issue:

```xml
<PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" ExcludeAssets="all" />
```

**Comment by teo-tsirpanis** [comment]
This package is very old (~~latest~~ last version released on November 2016) and totally unnecessary for sufficiently new TFMs (.NET Framework 4.7.1+, .NET Standard and .NET Core everything). It should not be referenced by projects targeting these frameworks. For your case, just let the compiler import the `RuntimeInformation` class from `mscorlib`.

**Comment by AArnott** [comment]
Thanks, @teo-tsirpanis. If I'm consuming another package that depends on it though, it causes issues. That's the case here. And if that other package actually targets net47, then it's justifiable that they have such a package reference. IMO this should Just Work. It cost me perhaps 30 minutes of investigation to understand what's going on and craft the workaround, and I'm a relative expert compared to most customers. I don't know how they would figure out how to get unblocked.

**Comment by teo-tsirpanis** [comment]
OK, I see your point now. I thought that package was from VS.

**Comment by tannergooding** [comment]
CC. @ericstj, @dotnet/area-infrastructure-libraries should this be reassigned? I don't think there is much the area-System.Runtime owners can do here since it seems infra/packaging related

**Comment by ericstj** [comment]
> I have a .NET Framework 4.7.2 project that references mscorlib and System.Runtime.InteropServices.RuntimeInformation as follows:

> C:\NugetCache\VS.ExternalAPIs.NetFX.V4_7_2.1.0.18091101\ref\mscorlib.dll
> C:\Users\andarno.nuget\packages\system.runtime.interopservices.runtimeinformation\4.0.0\ref\netstandard1.1\System.Runtime.InteropServices.RuntimeInformation.dll

That same framework (4.7.2) contains a higher version of `System.Runtime.InteropServices.RuntimeInformation.dll`
```
"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.InteropServices.RuntimeInformation.dll"
```
When packages are "absorbed" by a framework we don't go back and ship new versions of those packages for a new framework.  Instead we recommend that folks move to the new framework and remove their dependency on the package that is no longer needed.

I would expect MSBuild / SDK to prefer the version in the framework.  Indeed if I create a simple project targeting .NET 4.7.2 (either classic CSProj or new-style SDK project) I see that it does prefer the version in the framework and removes the one from the package.

This happens as part of the `_HandlePackageFileConflicts` target which lives in the following targets for classic CSProj
```
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\Microsoft.NET.Build.Extensions\Microsoft.NET.Build.Extensions.ConflictResolution.targets
```
And these targets in SDK-projects
```
C:\Program Files\dotnet\sdk\6.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ConflictResolution.targets
```

cc @dsplaisted 

@AArnott -- perhaps something about your build system is not running these targets?  If you can produce a simple repro that exhibits the problem outside of a custom build system then it's probably a bug in dotnet/sdk.

**Comment by AArnott** [comment]
That may explain why I couldn't repro the problem outside one specific project in a repo with old and custom targets. Given what's been explored here, and that I have a workaround, I'll close the bug.
Thanks.
